Features/nginx config playbook

adrian-arapiles opened the pull request:
### Description of the Change

- Add possibility to override nginx.conf from playbook.
- Fix wrong configuration of proxy_buffer in nginx.conf
- Fix security alert of ansible version. Update all test-requirements.txt.

### Benefits
Can override nginx.conf for custimize options
Well configure buffer-size option.
### Possible Drawbacks

None that i know of.

### Applicable Issues

[#62](https://github.com/idealista/nginx_role/issues/62)

dortegau committed:
Merge pull request #35 from idealista/develop

Develop

Commit:
Merge pull request #44 from idealista/develop

1.7.0 to master

Commit:
Merge pull request #50 from idealista/develop

v1.8.0 to master

jmonterrubio committed:
Merge pull request #55 from idealista/develop

Develop

Commit:
Merge pull request #61 from idealista/develop

v2.0.1 to master

adrian-arapiles committed:
Add possibility to override nginx.conf from playbook.

adrian-arapiles committed:
Fix wrong configuration of proxy_buffer in nginx.conf

adrian-arapiles committed:
Fix security alert of ansible version. Update all test-requirements.txt.
Molecule test work well.

dortegau commented on CHANGELOG.md:
```diff
@@ -1,108 +1,112 @@
-# Change Log
```
Why has the entire file been changed?

adrian-arapiles replied:
The file was commited with CRLF end line character. I commit with LF end line.

dortegau commented on tasks/config.yml:
```diff
@@ -20,7 +20,7 @@
 - name: NGINX | Copy server config
   template:
-    src: "{{ item }}.j2"
+    src: "{{ nginx_conf_template_path }}/{{ item }}.j2"
```
I think this won’t work with default (empty) ‘nginx_conf_template_path’

adrian-arapiles replied:
I moved the nginx.conf.j2 to templates/nginx folder.

dortegau commented on test-requirements.txt:
```diff
@@ -1,3 +1,3 @@
-molecule>=2.20.0,<2.21.0
-docker>=3.7.0,<3.8.0
\ No newline at end of file
+ansible==2.8.4
```
This change affects versions described in Readme file and meta/main.yml file

adrian-arapiles replied:
I changed both files now with correct version of ansible.

miguel-chacon commented on defaults/main.yml:
```diff
@@ -34,6 +34,7 @@ nginx_access_log: "{{ nginx_log_path }}/access.log"
 nginx_error_log: "{{ nginx_log_path }}/error.log"
 nginx_conf_path: "/etc/nginx"
 nginx_conf: "{{ nginx_conf_path }}/nginx.conf"
+nginx_conf_template_path: ''
```
This variable cannot be empty. If you don't override the variable with some existing path it will fail, it must be 'templates'

adrian-arapiles replied:
Fixed, thanks! :)

miguel-chacon commented on tasks/config.yml:
```diff
@@ -20,7 +20,7 @@
 - name: NGINX | Copy server config
   template:
-    src: "{{ item }}.j2"
+    src: "{{ nginx_conf_template_path }}/{{ item }}.j2"
```
This step fails if `nginx_conf_template_path` = ''

adrian-arapiles committed:
Fix nginx.conf template path defaults.

dortegau commented on defaults/main.yml:
```diff
@@ -34,6 +34,7 @@ nginx_access_log: "{{ nginx_log_path }}/access.log"
 nginx_error_log: "{{ nginx_log_path }}/error.log"
 nginx_conf_path: "/etc/nginx"
 nginx_conf: "{{ nginx_conf_path }}/nginx.conf"
+nginx_conf_template_path: 'templates'
```
This makes no sense in a Playbook using multiple roles (a common use case), because this will takes all the files included in the `templates` directory, even if that files aren't for configure nginx

adrian-arapiles committed:
Update meta and readme with upgrade of ansible version used.
Moved nginx.conf to nginx folder on templates. Update default path to this folder.

dortegau commented on defaults/main.yml:
```diff
@@ -34,6 +34,7 @@ nginx_access_log: "{{ nginx_log_path }}/access.log"
 nginx_error_log: "{{ nginx_log_path }}/error.log"
 nginx_conf_path: "/etc/nginx"
 nginx_conf: "{{ nginx_conf_path }}/nginx.conf"
+nginx_conf_template_path: 'templates/nginx'
```
As we are using double quotes as a De facto standard, I would prefer to use double quotes here

adrian-arapiles committed:
Use double quotes as other variables

dortegau commented on templates/nginx/nginx.conf.j2:
```diff
@@ -0,0 +1,48 @@
+    tcp_nopush on;
+
+    log_format  main '$remote_addr - [$time_local] ($request_time) $status '
+    '"$request" $request_length $body_bytes_sent '
```
There is an unneeded space after the last single quote

adrian-arapiles replied:
The space meaning that the format continue on next line. This format is the same at the documentation https://docs.nginx.com/nginx/admin-guide/monitoring/logging/#setting-up-the-access-log

dortegau replied:
In that case I would prefer to add indentation after this lines (line 40 and 41)

adrian-arapiles committed:
Indented format of nginx log for better readability

dortegau commented:
⚠️ Destination branch should be `idealista:develop` not `idealista:master`
